### DISCUSSION TITLE
issue: Rules of useController not reacting to changes

### DISCUSSION THREAD

**Discussion by ykolloch** [react-hook-form/react-hook-form#8312] [Q&A] [answered]
### Version Number

7.30.0

### Codesandbox/Expo snack

https://codesandbox.io/s/controller-rules-forked-6isro7?file=/src/App.tsx

### Steps to reproduce

Hey great lib! Only got one small issue,
if i use `useController` on my fields, the rules don't update on changes.

Like in the codesandbox shown. The error text below the input doesn't get removed when the state changes and the field is no longer required.

As a workaround we unregistered the fields now if the rules change, which works. But we were wondering if this is the intended behaviour. From this [thread](https://github.com/react-hook-form/react-hook-form/issues/1749) we got the impression that from V6 on rules will update on render.

### Expected behaviour

Rules and errors in `useController` fields update on render.

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
It's working as expected, your error message doesn't get removed because user is not interacted with the field before the errors gets removed, if you want to remove when you toggle, you can consider to use `trigger`

**Reply by ykolloch** [reply]
Thanks for your quick response. 

We tried your solution with using `trigger` and it works like a charm. Thank your for that also 👍 

I think we can close this discussion/issue. 
Have a nice weekend

**Comment by bluebill1049** [comment]
https://user-images.githubusercontent.com/10513364/167034879-cde84817-6f1a-4d1d-bf68-90588b4d2102.mov

the above is expected and current behavior.
